noxfile.py

```diff
@@ -146,26 +146,23 @@ def generate_test_data(session: Session) -> None:
 @nox.session(name="pip-compile")
 def pip_compile(session: Session) -> None:
     """Update requirements.txt and requirements-extras.txt files."""
-    PWD = session.env["PWD"]
-    PYTHON_VERSION_MINIMAL = parse_supported_python_versions()[0]
-    # git must be installed in the image due to setuptools-scm that has it as a direct dependency
-    pip_compile_cmd = (
-        "apk add git && "
-        "pip3 install pip-tools && "
-        "pip-compile --generate-hashes --output-file=requirements.txt --rebuild pyproject.toml && "
-        "pip-compile --all-extras --generate-hashes --output-file=requirements-extras.txt --rebuild pyproject.toml"
-    )
+    REQUIRED_PYTHON_VERSION = parse_supported_python_versions()[0]
+    session.install("uv")
     cmd = [
-        "podman",
-        "run",
-        "--rm",
-        "--volume",
-        f"{PWD}:/cachi2:rw,Z",
-        "--workdir",
-        "/cachi2",
-        f"docker.io/library/python:{PYTHON_VERSION_MINIMAL}-alpine",
```

Comment:
I'm afraid you won't be able to get rid of the container just yet, because its sole purpose was to shield one from having to install python 3.9 on their devel machine just to be able to run pip-compile. Unit tests and linters are tested with different versions of python so we're covered both in upstream CI and locally.